product-service/src/utils/errorHandler.js

```diff
@@ -0,0 +1,16 @@
+import {NotFoundError} from "./NotFoundError";
+import {BadRequest} from "./BadRequest";
+import {createResponse} from "./createResponse";
+
+export const errorHandler = (error) => {
+    console.log(`error in errorHandler ${JSON.stringify(error)}`);
+
+    if (error instanceof NotFoundError) {
+        return createResponse(404, error.message);
+    }
+    if (error instanceof BadRequest) {
```

Comment:
checking of the instances' classes goes against solid

Reply:
Thanks for review. SOLID includes 5 statements and your assertion looks a little surface. Which solid statement does it ruin? The same approach we use when checking objects against interfaces. Maybe you have something to propose ?

Reply:
Hi, the Liskov Substitution Principle restricts class instance type checking.